src/derivative_operators/convolutions.jl

```diff
@@ -43,92 +60,193 @@ end
 function convolve_BC_right!(x_temp::AbstractVector{T}, x::AbstractVector{T}, A::DerivativeOperator) where {T<:Real}
     stencil = A.high_boundary_coefs
     coeff   = A.coefficients
-    for i in 1 : A.boundary_point_count
-        cur_stencil = stencil[i]
+    N       = length(x)
+    L       = A.boundary_stencil_length
+
+    _bpc = A.boundary_point_count
+    flag = false
+
+    if stencil == []
```

Comment:
isempty

Reply:
@ChrisRackauckas  This has been fixed in #130. Sorry for the confusion but these two PRs cannot be separated. The changes here were needed there. So should we separately merge this and Upwind Operators or just merge the Upwind?